### ISSUE TITLE
Clarify benefits of Godot's In-Editor IDE

### ISSUE THREAD

**Issue by ShalokShalom** [Godot-Languages-Support/godot-lang-support#11]
Anything else?

![Screenshot_2020-09-03-19-02-44-20](https://user-images.githubusercontent.com/6344099/92145575-b5815800-ee18-11ea-9493-640dfc67a2ed.jpg)

**Comment by Vivraan** [comment]
These are not available in Python or Lua bindings, yes?

**Comment by perbone** [comment]
I may be missing something here but my Lua implementation has all the functionality that the api provides and is on par with GDScript.

**Comment by ShalokShalom** [comment]
> These are not available in Python or Lua bindings, yes?

They are.

**Comment by ShalokShalom** [comment]
> I may be missing something here but my Lua implementation has all the functionality that the api provides and is on par with GDScript.

Yeah, this is how I also understood it. Now is it on us, to clarify for the reader, what that is. An own section with in-detail description - an animated PNG - could help to illustrate the difference in user experience and to list all the benefits. 

Congrats 🥳

**Comment by Vivraan** [comment]
That'd actually be amazing - especially since it'll very clearly map visually to what people might be expecting!

**Comment by ShalokShalom** [comment]
One of those techniques can be used: https://ezgif.com/help/alternative-animated-image-formats

**Comment by Vivraan** [comment]
Okay, basically upload a gif of a speeded up version of operating in-editor and outside-editor. Sounds good.

**Comment by ShalokShalom** [comment]
Yeah, just that GIF are really outdated, I love the solutions linked. GIFs provide some very convenient features, while other solutions provide much improved image quality.

**Comment by Vivraan** [comment]
I use ezgif myself hehe

**Comment by Vivraan** [comment]
Is the current document complying with these observations?

**Comment by ShalokShalom** [comment]
How you mean? If that is already implemented? No, I dont think so?

**Comment by Vivraan** [comment]
Alright, we'll need to provide a clear description of what *Godot Editor Integrated Support* means.

**Comment by perbone** [comment]
Basically, you have the experience of a source code editor, similar to VS Code, for example, with features like syntax highlighting, code complete, syntax and static analysis for errors and warnings and code execution debug. So the net result is a streamlined user expirience with all the capabilities availble without compromises. All of this is accomplished through the use of the Script API.

**Comment by Vivraan** [comment]
Agreed! Although I'd really like some of the stuff I've come to expect from Electron based Code Editors such as multi-cursor selection and editing, renaming variables semantically (without using Find and Replace), etc. Afaik these are not part of the Godot Code Editor.

I'd assume the corollary for Visual Script is something similar?

**Comment by ShalokShalom** [comment]
We could list the benefits of the Godot editor and that one of the external plugins, all in a spreadsheet. So all people see at a glance, what plugin supports which feature. Editor specific features could be implemented, so has any external plugin in VSCode automatically the bespoken features like multi cursors and so on. What do you think about it?

**Comment by Vivraan** [comment]
> Editor specific features could be implemented
What do you mean 🤔?

**Comment by ShalokShalom** [comment]
> > Editor specific features could be implemented
> 
> What do you mean 🤔?

All editors - like the build-in one and any external one like VSCode - provide unique features, as you pointed out.

I suggest to list them in the spreadsheet.

**Comment by Vivraan** [comment]
The document seems to need a major redo based on the current suggestions. 😅

**Comment by ShalokShalom** [comment]
Just an added spreadsheet.

**Comment by Vivraan** [comment]
@ShalokShalom 

| like this | ? |
|----------|--|
| this is | okay ig |

**Comment by ShalokShalom** [comment]
Yes
